src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -269,33 +311,56 @@ export class MeetingManager implements AudioVideoObserver {
       (await this.audioVideo?.listAudioOutputDevices()) || [];
   }
 
-  setupDeviceLabelTrigger(): void {
-    const callback = async (): Promise<MediaStream> => {
-      this.devicePermissionStatus = DevicePermissionStatus.IN_PROGRESS;
-      this.publishDevicePermissionStatus();
-      try {
-        const devices = await navigator.mediaDevices.enumerateDevices();
-        const hasVideoInput = devices.some(value => {
-          return value.kind === 'videoinput';
-        });
-
-        const stream = await navigator.mediaDevices.getUserMedia({
-          audio: true,
-          video: hasVideoInput
-        });
+  async updateDeviceList(device: Device): Promise<void> {
```

Comment:
Isn't this a breaking changes? Before builders can call `meetingManager.updateDeviceList()` but not it would fail.

Reply:
No, it's not a breaking change. This is a new function, the original one is called `updateDeviceLists()`, it's unchanged. 
I was wondering should I merge these two functions? e.g. make the updateDeviceLists() as default behavior in switch case, or something else.

Reply:
This is confusing in API levels to have 2 API so similar name. You should rewrite the old one and add device as an optional parameter

Reply:
Or make the new one supersede the old one. Getting in the habit of returning `Promise` is a good thing.

Reply:
The `DeviceKind` enum only has `AudioInput`, `AudioOutput` and `VideoInput` now. So the `updateDeviceList()` is designed to support device specific operation. If I need to merge the `updateDeviceLists()`(the original one) in to this one, it should be better to have some other option in this enum.

How about add `AuidoAndVideo` and `None` in `DeviceKind` enum? And if we do so, the `DeviceKind` enum will almost be a superset of `DeviceLabels`. Should we keep both of them?

```ts
export enum DeviceLabels {
  None,
  Audio,
  Video,
  AudioAndVideo,
};

export enum DeviceKind {
  AudioInput,
  AudioOutput,
  VideoInput,
}
```